src/Collections-Strings/String.class.st

```diff
@@ -883,7 +895,8 @@ String >> compare: string1 with: string2 collated: order [
 	"('a000' asWideString sameAs: (ByteArray with: 97 with: 0 with: 0 with: 0) asString) >>> false"
 	
 	(string1 isByteString and: [string2 isByteString]) ifTrue: [
-		^ ByteString compare: string1 with: string2 collated: order].
+		^ (string1 compareWith: string2 collated: order) sign + 2
```

Comment:
Could we add a comment on why the "+2"? :)

Reply:
Of course, I'll do that on the reworked PR I'm doing with pablo